mayastor/src/bdev/nexus/nexus_share.rs

```diff
@@ -112,188 +85,87 @@ impl Share for Nexus {
+            return Err(Error::AlreadyShared {
                 name: self.name.clone(),
-            })?
-        } else {
-            self.name.clone()
-        };
-
-        debug!("creating share handle for {}", name);
-        // The share handle is the actual bdev that is shared through the
-        // various protocols.
+            });
+        }
 
-        let device_id = match share_protocol {
+        match protocol {
             ShareProtocolNexus::NexusNbd => {
-                // Publish the nexus to system using nbd device and return the
-                // path to nbd device.
-                let nbd_disk =
-                    NbdDisk::create(&name).await.context(ShareNbdNexus {
-                        name: self.name.clone(),
-                    })?;
-                let device_uri = nbd_disk.as_uri();
-                self.nexus_target = Some(NexusTarget::NbdDisk(nbd_disk));
-                device_uri
-            }
-            ShareProtocolNexus::NexusIscsi => {
-                // Publish the nexus to system using an iscsi target and return
-                // the IQN
-                let iscsi_target = NexusIscsiTarget::create(&name).context(
-                    ShareIscsiNexus {
+                let disk = NbdDisk::create(&self.name).await.context(
+                    ShareNbdNexus {
                         name: self.name.clone(),
                     },
                 )?;
-                let uri = iscsi_target.as_uri();
-                self.nexus_target =
-                    Some(NexusTarget::NexusIscsiTarget(iscsi_target));
-                uri
+                let uri = disk.as_uri();
+                self.nexus_target = Some(NexusTarget::NbdDisk(disk));
+                Ok(uri)
+            }
+            ShareProtocolNexus::NexusIscsi => {
+                let uri = self.share_iscsi().await?;
+                self.nexus_target = Some(NexusTarget::NexusIscsiTarget);
+                Ok(uri)
             }
             ShareProtocolNexus::NexusNvmf => {
-                let nvmf_target = NexusNvmfTarget::create(&name)
-                    .await
-                    .context(ShareNvmfNexus {
-                        name: self.name.clone(),
-                    })?;
-                let uri = nvmf_target.as_uri();
-                self.nexus_target =
-                    Some(NexusTarget::NexusNvmfTarget(nvmf_target));
-                uri
+                let uri = self.share_nvmf().await?;
+                self.nexus_target = Some(NexusTarget::NexusNvmfTarget);
+                Ok(uri)
             }
-        };
-        self.share_handle = Some(name);
-        Ok(device_id)
+        }
     }
 
-    /// Undo share operation on nexus. To the chain of bdevs are all claimed
-    /// where the top-level dev is claimed by the subsystem that exports the
-    /// bdev. As such, we must first destroy the share and move our way down
-    /// from there.
```

Comment:
did you mean to replace this with another comment?

Reply:
This pertains to the bdev encryption code that has been removed (and the comments with it) for now.